matching-core/src/main/java/com/graphhopper/matching/JsonFile.java

```diff
@@ -0,0 +1,70 @@
+package com.graphhopper.matching;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.graphhopper.util.PointList;
+
+public class JsonFile {
+	protected MatchResult result;
+	
+	public JsonFile(MatchResult result) {
+		this.result = result;
+	}
+
+	public void exportTo(String outFile) {
+		try {
+			File file = new File(outFile);
+			if (!file.exists()) {
+				file.createNewFile();
+			}
+			this.exportTo(new PrintWriter(outFile));			
+		} catch(IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	public void exportTo(PrintWriter writer) {
+		JSONObject root = new JSONObject();
+		JSONObject diary = new JSONObject();
+		JSONArray entries = new JSONArray();
+		JSONObject route = new JSONObject();
+		JSONArray links = new JSONArray();
+		for (int emIndex = 0; emIndex < result.getEdgeMatches().size(); emIndex++) {
+			JSONObject link = new JSONObject();
+			JSONObject geometry = new JSONObject();
+			geometry.put("type", "LineString");
+			
+			EdgeMatch edgeMatch = result.getEdgeMatches().get(emIndex);
+			PointList pointList = edgeMatch.getEdgeState().fetchWayGeometry(emIndex == 0 ? 3 : 2);
+			geometry.put("coordinates", pointList.toGeoJson());
+			
+			link.put("src", edgeMatch.getEdgeState().getEdge());
```

Comment:
@karussell thanks, I will check if I will need the store the OSM id at this moment.